File: CS50/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    //printf("%s\n", text);
    //printf("%i letter(s)\n", count_letters(text));
    //printf("%i word(s)\n", count_words(text));
    //printf("%i sentence(s)\n", count_sentences(text));

    //Coleman-Liau readability index = 0.0588 * L - 0.296 * S - 15.8
    float L = count_letters(text) / (float) count_words(text) * 100;
    float S = count_sentences(text) / (float) count_words(text) * 100;

    //printf("L: %f , S: %f\n", L, S);
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index > 1 && index < 16)
    {
        printf("Grade %i\n", index);
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}

int count_letters(string text)
{
    int letters = 0;
    //cheking if character in the range of A - z
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isupper(text[i]) || islower(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 0;
    //counting words only by checking for space between
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    return words + 1;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]) && (text[i - 1] == '.' || text[i - 1] == '?' || text[i - 1] == '!'))
        {
            sentences++;
        }
    }
    return sentences + 1;
}

File: CS50/readability.py
from cs50 import get_string

text = get_string("Text: ")

# Initialize variables for counting
letters = 0
words = 0
sentences = 0

for i in range(len(text)):
    # If alphanumeric character increment letters
    if text[i].isalpha():
        letters += 1
    # If space increment words
    elif text[i].isspace():
        words += 1
        # And if the previous character is ".", "?", "!", increment sentences
        if text[i - 1] == "." or text[i - 1] == "?" or text[i - 1] == "!":
            sentences += 1

# Increment by 1 for the last word and sentence that doesn't have space
words += 1
sentences += 1

# Coleman-Liau readability index = 0.0588 * L - 0.296 * S - 15.8
L = letters / words * 100
S = sentences / words * 100

index = round(0.0588 * L - 0.296 * S - 15.8)
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
